### PR TITLE
initial version of sumologic-fluentd

### PR DESCRIPTION
**Pull request by flah00**
Initial version of a sumologic and fluent solution. It provides more flexibility than the sumologic collector, because it easily integrates environment variables. This approach also removes the need for keys and secrets. This is replaced with an HTTP collector, managed in the sumo console.

The chart provides some value adds, over the simple [daemon set](https://github.com/SumoLogic/fluentd-kubernetes-sumologic/blob/master/fluentd.daemonset.yaml) the project provides. Perhaps the most significant is the option to persist the fluent pos files, on each node. It also very neatly exposes all of the options available to the image and codifies the management of the collector url secret.

## Limitations

The current docker image, for [fluentd-kubernetes-sumologic](https://github.com/SumoLogic/fluentd-kubernetes-sumologic)
does not support systemd. The only logs available to SumoLogic are those in
`/var/log/containers`. Logs generated in the `kube-system` namespace can only be
excluded using `EXCLUDE_NAMESPACE`, but there's no finer granularity for containers in the namespace. The `EXCLUDE_PATH` option is not relevant on a systemd OS. I'm working with the maintainers, to find a [solution](https://github.com/SumoLogic/fluentd-kubernetes-sumologic/issues/19).

### CODE REVIEW

**Comment by flah00** [comment]
@seanknox maintainer of sumokube

File: stable/sumologic-fluentd/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: sumologic-fluentd
+version: 0.1.0
+description: Sumologic Log Collector
+keywords:
+  - monitoring
+  - logging
+sources:
+  - https://github.com/SumoLogic/fluentd-kubernetes-sumologic
+maintainers:
+  - name: Philip Champon
```
Comment (prydonius) [inline stable/sumologic-fluentd/Chart.yaml]: Can you change this to `flah00`? We're starting to use GitHub usernames here instead.

File: stable/sumologic-fluentd/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name: sumologic-fluentd
+version: 0.1.0
+description: Sumologic Log Collector
+keywords:
+  - monitoring
+  - logging
+sources:
+  - https://github.com/SumoLogic/fluentd-kubernetes-sumologic
```
Comment (prydonius) [inline stable/sumologic-fluentd/Chart.yaml]: I would set this to the `home` as well and add the sumologic icon

File: stable/sumologic-fluentd/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 -}}
```
Comment (prydonius) [inline stable/sumologic-fluentd/templates/_helpers.tpl]: Please add `| trimSuffix "-"` to the end of this

File: stable/sumologic-fluentd/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 64 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "fullname" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 -}}
```
Comment (prydonius) [inline stable/sumologic-fluentd/templates/_helpers.tpl]: same as above

File: stable/sumologic-fluentd/templates/daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  name: {{ template "fullname" . }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  template:
+    metadata:
+      name: {{ template "fullname" . }}
+      labels:
+        app: {{ template "fullname" . }}
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (prydonius) [inline stable/sumologic-fluentd/templates/daemonset.yaml]: Having these labels in the Pod template can cause issues when upgrade the DaemonSet as the selector changes. You should either only keep `app` or set a custom selector.

File: stable/sumologic-fluentd/templates/daemonset.yaml [hunk 6]
```diff
@@ -0,0 +1,112 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (prydonius) [inline stable/sumologic-fluentd/templates/daemonset.yaml]: missing app, release and heritage labels

File: stable/sumologic-fluentd/templates/daemonset.yaml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  template:
+    metadata:
+      name: {{ template "fullname" . }}
+      labels:
+        app: {{ template "fullname" . }}
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      serviceAccount: {{ template "fullname" . }}
```
Comment (prydonius) [inline stable/sumologic-fluentd/templates/daemonset.yaml]: this should be `serviceAccountName`. `serviceAccount` is a deprecated alias

File: stable/sumologic-fluentd/values.yaml [hunk 8]
```diff
@@ -0,0 +1,79 @@
+# Default values for sumologic-fluentd.
+image:
+  name: sumologic/fluentd-kubernetes-sumologic
+  tag: latest
+  pullPolicy: IfNotPresent
```
Comment (prydonius) [inline stable/sumologic-fluentd/values.yaml]: `pullPolicy` should be `Always` for `latest` tag

**Review by prydonius — CHANGES_REQUESTED** [review]
Thanks @flah00, sorry for the late review!

**Comment by flah00** [comment]
@prydonius I believe all issues have been addressed, thanks for the review!

**Review by prydonius — APPROVED** [review]

**Comment by prydonius** [comment]
code review looks good, @seanknox would you be able to UX review this?

**Comment by flah00** [comment]
@seanknox can i get a you-ex? :)

**Comment by flah00** [comment]
@prydonius could we mebs get a diff ux reviewer? i've got a pending patch to this chart, for systemd which i would like to also submit ... and since this has already been approved, i was thinking it'd be bad form to piggy back on this pr.

**Comment by flah00** [comment]
@prydonius SumoLogic's very own @frankreno has graciously agreed to participate. I updated the `Chart.yaml`. What's next?

**Comment by frankreno** [comment]
feel free to add my email - freno@sumologic.com to the maintainers as well.

**Review by prydonius — APPROVED** [review]

**Comment by prydonius** [comment]
Awesome - thanks @flah00 and @frankreno :)

**Pull request merged**